### PR TITLE
Use common function signature for instruction implementations

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by gumb0** [comment]
Shouldn't `OP_GAS` in Baseline now use `gas()` function?

**Comment by chfast** [comment]
> Shouldn't `OP_GAS` in Baseline now use `gas()` function?

Fixed.

File: lib/evmone/instructions.cpp [hunk 1]
```diff
@@ -10,24 +10,10 @@ namespace evmone
 {
 namespace
 {
-template <void InstrFn(Stack&)>
+template <InstrFn Fn>
```
Comment (gumb0) [inline lib/evmone/instructions.cpp:13]: Can't you leave the signature here?
```suggestion
template <template <evmc_status_code Fn(ExecutionState&)>>
```

Comment (chfast) [reply lib/evmone/instructions.cpp:13]: I can. This however anticipates another change where the table of instruction implementations is introduced: https://github.com/ethereum/evmone/commit/1fade9cca2024398a9be8bf66016bfc578c72fd4. The table itself is not needed yet, so I don't wanted to commit it early.

Do you think it is better to leave the `InstrFn` for later?

Comment (gumb0) [reply lib/evmone/instructions.cpp:13]: Ok to do it later.

**Review by gumb0 — APPROVED** [review]

**Pull request merged**